File: source/cubing/utils/Reflections.java
package cubing.utils;

import java.lang.reflect.*;

public class Reflections
{
    public Reflections() {
        super();
    }
    
    public static void packetSetValue(final Object obj, final String name, final Object value) {
        try {
            final Field field = obj.getClass().getDeclaredField(name);
            field.setAccessible(true);
            field.set(obj, value);
        }
        catch (final Exception exception) {
            exception.printStackTrace();
        }
    }
    
    public static Object packetGetValue(final Object obj, final String name) {
        try {
            final Field field = obj.getClass().getDeclaredField(name);
            field.setAccessible(true);
            return field.get(obj);
        }
        catch (final Exception exception) {
            exception.printStackTrace();
            return null;
        }
    }
}

File: src/main/kotlin/cn/cyanbukkit/fastbuild/utils/Reflections.kt
package cn.cyanbukkit.fastbuild.utils

object Reflections {

    fun packetSetValue(obj: Any, name: String, value: Any) {
        try {
            val field = obj::class.java.getDeclaredField(name)
            field.isAccessible = true
            field.set(obj, value)
        } catch (exception: Exception) {
            exception.printStackTrace()
        }
    }

    fun packetGetValue(obj: Any, name: String): Any? {
        return try {
            val field = obj::class.java.getDeclaredField(name)
            field.isAccessible = true
            field.get(obj)
        } catch (exception: Exception) {
            exception.printStackTrace()
            null
        }
    }

}